File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeDeepLinkActivity.java
package com.braintreepayments.api;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * This Activity is used by the SDK Activity to handle deep links on behalf of the merchant
 * It should not be used directly.
 */
public class BraintreeDeepLinkActivity extends AppCompatActivity {

    private final BrowserSwitchClient browserSwitchClient = new BrowserSwitchClient();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        browserSwitchClient.captureResult(this);
        finish();
    }
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeDeepLinkActivity.kt
package com.braintreepayments.api

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * This Activity is used by the SDK Activity to handle deep links on behalf of the merchant
 * It should not be used directly.
 */
// NEXT_MAJOR_VERSION : remove this class
class BraintreeDeepLinkActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        finish()
    }
}
